src/Shared/ITranslator.cs

```diff
@@ -259,6 +259,18 @@ void Translate<T>(ref T value)
         /// <param name="byteArray">The array to be translated.</param>
         void Translate(ref byte[] byteArray);
 
+        
+        /// <summary>
+        /// Translates a byte array
+        /// </summary>
+        /// <param name="byteArray">The array to be translated.</param>
+        /// <param name="length">The length of array which will be used in translation</param>
+        /// <remarks>
+        /// This method should be used only for serialization, not for reading from stream.
+        /// Typical usage will be for example in combination with <see cref="MemoryStream.GetBuffer"> when we get array, but we want to use only part of it.
+        /// </remarks>
+        void Translate(ref byte[] byteArray, int length);
```

Comment:
Just thinking out loud: Maybe if the signature was:

```
void Translate(ref byte[] byteArray, ref int length);
```

you wouldn't have to comment that `length` is used only for writing. Reading would populate the argument with the length of the array (not super useful, I know, but eliminates this inelegant pattern where the interface method has a parameter that we know is going to be unused in half of its implementations). What do you think?